reconcile/dashdotdb_dvo.py

```diff
@@ -0,0 +1,126 @@
+DASHDOTDB_SECRET = os.environ.get('DASHDOTDB_SECRET',
+                                  'app-sre/dashdot/auth-proxy-production')
+
+
+class DashdotdbDVO:
+    def __init__(self, dry_run, thread_pool_size):
+        self.dry_run = dry_run
+        self.thread_pool_size = thread_pool_size
+        self.settings = queries.get_app_interface_settings()
+        secret_reader = SecretReader(settings=self.settings)
+        secret_content = secret_reader.read_all({'path': DASHDOTDB_SECRET})
+        self.dashdotdb_url = secret_content['url']
+        self.dashdotdb_user = secret_content['username']
+        self.dashdotdb_pass = secret_content['password']
+        self.logmarker = "DDDB_DVO:"
+
+    def _post(self, deploymentvalidation):
+        if deploymentvalidation is None:
+            return None
+        cluster = deploymentvalidation['cluster']
+        dvdata = deploymentvalidation['data']
+        LOG.info('%s posting validations for %s', self.logmarker, cluster)
+        response = None
+        if not self.dry_run:
+            endpoint = (f'{self.dashdotdb_url}/api/v1/'
+                        f'deploymentvalidation/{cluster}')
+            response = requests.post(url=endpoint, json=dvdata,
+                                     auth=(self.dashdotdb_user,
+                                           self.dashdotdb_pass))
+            try:
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as details:
+                LOG.error('%s error posting %s - %s',
+                          self.logmarker, cluster, details)
+
+        LOG.info('%s cluster %s synced', self.logmarker, cluster)
+        return response
+
+    def _promget(self, url, query, token=None):
+        uri = 'api/v1/query'
+        url = urljoin('https://'+url, uri)
+        headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer " + token,
+        }
+        params = {'query': query}
+        LOG.info('%s Fetching prom payload from %s?%s',
+                 self.logmarker, url, query)
+        response = requests.get(url,
+                                params=params,
+                                headers=headers,
+                                timeout=(5, 30))
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as details:
+            LOG.error('%s error accessing prometheus - %s',
+                      self.logmarker, details)
+
+        response = response.json()
+        # TODO ensure len response == 1
+        # return response['data']['result']
+        return response
+
+    def _get_automationtoken(self, cluster):
+        autotoken_reader = SecretReader(settings=self.settings)
+        autotoken = {'path': "app-sre/creds/kube-configs/" + cluster,
+                     'field': "token"}
+        token = autotoken_reader.read(autotoken)
+        return token
+
+    def _get_deploymentvalidation(self, cluster, validation, oc_map):
```

Comment:
I don't see where you're using `oc_map` in this function.

Reply:
oc_map used by the threaded iteration here [1], looked like this was "the way" iterating over the clusters was being done elsewhere.
[1] https://github.com/mmclanerh/qontract-reconcile/blob/b8d5b0f3eeb81de5d6e1e8c0895759089c39150c/reconcile/dashdotdb_dvo.py#L114-L118

Reply:
You don't need `oc_map` in the `_get_deploymentvalidation` function. As you can see in https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/threaded.py#L17, `validation` and `oc_map` are the `**kwargs` that are passed to the `func`. You should be able to safely remove from here and from the `threaded` call

Reply:
yeah reviewing.. this was cruft. removed parameter.